Sources/BranchSDK/BranchContentDiscoverer.m

```diff
@@ -27,12 +27,11 @@ @implementation BranchContentDiscoverer
 
 + (BranchContentDiscoverer *)getInstance {
     static BranchContentDiscoverer *sharedInstance = nil;
-    @synchronized (self) {
-        if (!sharedInstance) {
-            sharedInstance = [[BranchContentDiscoverer alloc] init];
-        }
+    static dispatch_once_t onceToken = 0;
+    dispatch_once(&onceToken, ^{
```

Comment:
Is this by default on the main thread?

Reply:
`dispatch_once` doesn't switch threads or anything like that. 

It's a shorthand for locking around Singleton creation.
https://developer.apple.com/documentation/dispatch/1447169-dispatch_once

Reply:
I see, I saw the synchronized keyword and questioned if there was a threading assumption. But since dispatch once happens .. once, then this makes sense.